pypdf/_writer.py

```diff
@@ -799,12 +803,22 @@ def _update_field_annotation(
             da = da.get_object()
         font_properties = da.replace("\n", " ").replace("\r", " ").split(" ")
         font_properties = [x for x in font_properties if x != ""]
-        font_name = font_properties[font_properties.index("Tf") - 2]
-        font_height = float(font_properties[font_properties.index("Tf") - 1])
+        if font_name:
+            font_properties[font_properties.index("Tf") - 2] = font_name
+        else:
+            font_name = font_properties[font_properties.index("Tf") - 2]
+        font_height = (
+            font_size
+            if font_size >= 0
+            else float(font_properties[font_properties.index("Tf") - 1])
+        )
         if font_height == 0:
-            font_height = rct.height - 2
-            font_properties[font_properties.index("Tf") - 1] = str(font_height)
-            da = " ".join(font_properties)
+            if field.get("/Ff", 0) & InteractiveFormDictEntries.Ff_Multiline:
+                font_height = 12
```

Comment:
Why 12?

Reply:
This is a default size I've identified using Acrobat.
This is to prevent very large fonts in case of multiline where you cannot guess which font height to use.
I will set is as a constant to be clerarer

Reply:
OK for you ?

Reply:
The default size in which case? For a specific form? Allowing the user to configure this sounds more suitable here.

Reply:
You can if you give a positive number